salt/states/pkg.py

```diff
@@ -2678,6 +2678,18 @@ def mod_aggregate(low, chunks, running):
                     elif 'name' in chunk:
                         version = chunk.pop('version', None)
                         if version is not None:
+                            if version == 'latest':
+                                try:
+                                    version = __salt__['pkg.latest_version'](chunk['name'],
+                                                                             fromrepo=chunk.get('fromrepo'),
+                                                                             refresh=chunk.get('refresh'))
```

Comment:
I'll have to check at which point aggregation happens. On one hand, we don't want to add unnecessary refreshes, but on the other hand, if refresh wasn't explicitly passed then this would pass ``refresh=None`` to ``pkg.latest_version``, which would cause no refresh to be performed and could give an inaccurate version.